backend/src/main/java/meowhub/backend/jpa_buddy/User.java

```diff
@@ -0,0 +1,189 @@
+    private PrivacySetting postsPrivacy;
+
+    @NotNull
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @OnDelete(action = OnDeleteAction.RESTRICT)
+    @JoinColumn(name = "FRIENDS_PRIVACY_ID", nullable = false)
+    private PrivacySetting friendsPrivacy;
+
+    @NotNull
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @OnDelete(action = OnDeleteAction.RESTRICT)
+    @JoinColumn(name = "ROLE_ID", nullable = false)
+    private Role role;
+
+    @NotNull
+    @Builder.Default
+    @Column(name = "ACCOUNT_NON_LOCKED", nullable = false)
+    private Boolean accountNonLocked = false;
+
+    @NotNull
+    @Builder.Default
+    @Column(name = "CREDENTIALS_NON_EXPIRED", nullable = false)
+    private Boolean credentialsNonExpired = false;
+
+    @Column(name = "CREDENTIALS_EXPIRY_DATE")
+    private LocalDate credentialsExpiryDate;
+
+    @Column(name = "CREATED_AT")
+    private LocalDate createdAt;
+
+    @Size(max = 36)
+    @Column(name = "CREATED_BY",length = 36)
+    private String createdBy;
+
+    @Column(name = "MODIFIED_AT")
+    private LocalDate modifiedAt;
+
+    @Size(max = 36)
+    @Column(name = "MODIFIED_BY", length = 36)
+    private String modifiedBy;
+
+    @OneToMany(mappedBy = "sender")
+    private final Set<Chatroom> chatroomsSender = new LinkedHashSet<>();
+
+    @OneToMany(mappedBy = "receiver")
+    private final Set<Chatroom> chatroomsReceiver = new LinkedHashSet<>();
+
+    @OneToMany(mappedBy = "author")
+    private final Set<ChatroomMessage> chatroomMessages = new LinkedHashSet<>();
+
+    @OneToMany(mappedBy = "user")
+    private final Set<Comment> comments = new LinkedHashSet<>();
+
+    @OneToMany(mappedBy = "user")
+    private final Set<GroupchatMessage> groupchatMessages = new LinkedHashSet<>();
+
+    @OneToMany(mappedBy = "user")
+    private final Set<MatchingProfile> matchingProfiles = new LinkedHashSet<>();
+
+    @OneToMany(mappedBy = "user")
+    private final Set<Picture> pictures = new LinkedHashSet<>();
+
+    @OneToMany(mappedBy = "user")
+    private final Set<Post> posts = new LinkedHashSet<>();
+
+    @OneToMany(mappedBy = "user")
+    private final Set<Profile> profiles = new LinkedHashSet<>();
+
+    @OneToMany(mappedBy = "user")
+    private final Set<UserGroup> userGroups = new LinkedHashSet<>();
+
+    @OneToMany(mappedBy = "sender")
+    private final Set<UserRelation> userRelationsSender = new LinkedHashSet<>();
+
+    @OneToMany(mappedBy = "receiver")
+    private final Set<UserRelation> userRelationsReceiver = new LinkedHashSet<>();
+
+    @OneToMany(mappedBy = "user")
+    private final Set<UserToken> userTokens = new LinkedHashSet<>();
+
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof User)) return false;
+        return id != null && id.equals(((User) o).getId());
+    }
+
+    @Override
+    public int hashCode() {
+        return getClass().hashCode();
+    }
```

Comment:
Why don't you use @EqualsAndHashcode annotation? If they're not custom then please replace them

Reply:
oh yeah, i thought u might look into it. As you can see, the equals function is overrided -> the object is only compared by its id. So i can't use @EqualsAndHashCode